### DISCUSSION TITLE
can't create Application.java file

### DISCUSSION THREAD

**Discussion by k0302** [firebase/flutterfire#4056] [Q&A]
## Bug report
1. I can't create Application.java file in the directory of java, since my project name is in the directory of kotlin.
   Is it OK to create Application.kt file in the directory of kotlin under where my project name is?

2. I'm also making a flutter web app. Can I use this plugin in my flutter web app, too?

**Comment by bdlukaa** [comment]
Which plugin and its version are you using?

**Comment by k0302** [comment]
I have firebase_messaging 6.0.13 and flutter_local_notifications 1.4.3 in pubspec.yaml. I also have 'com.google.firebase:firebase-messaging:21.0.0' in my app/build.gradle.

**Comment by bdlukaa** [comment]
upgrade your firebase_messaging to 8.0.0-dev.7. The plugin should work out of the box without any manual configuration for android, ios and macOS. Web is not supported.

**Comment by markusaksli-nc** [comment]
Hi @k0302 
`Application.kt` will work just fine. If you run into any issues try using the latest stable version `firebase_messaging: ^7.0.3`. The dev version should work out of the box but it is still under development and constantly subject to changes and improvements. There have already been a number of reported issues with the previous dev versions of `8.0.0`.
Does this solve the issue for you?
